src/BattleTest.java

```diff
@@ -0,0 +1,107 @@
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.function.Executable;
+import java.io.OutputStream;
+import java.io.PrintStream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class BattleTest {
+    private PrintStream printer;
+
+    @BeforeEach
+    public void printerSetup(){
+        printer = new PrintStream(new OutputStream() {
+                @Override
+                public void	write(int b) {
+                    // noop
+                }
+            });
+    }
+
+    @Test
+    @DisplayName("Test for battle function")
+    public void testForBattle1() {
+        // Arrange
+        Entity entity1 = new Entity(100, 50, 35, 2, "Bob", "Bob story");
+        Entity entity2 = new Entity(90, 40, 30, 2, "Mike", "Mike story");
+
+        Battle battle = new Battle(printer);
+
+        // //Act
+        battle.battle(entity1, entity2);
+
+        // Assert
+        assertTrue(entity1.getHealth() > entity2.getHealth(), "Wrong health");
```

Comment:
Nem ertem, miert ilyen atvitt tesztet csinalsz. Itt konkretan a vegeredmenyt is tudod, letesztelhetned azt. De nem egeto baj.